src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/X86/Bmi2.cs

```diff
@@ -79,7 +90,17 @@ public new abstract class X64 : X86Base.X64
         ///   MULX r32a, r32b, reg/m32
         /// The above native signature does not directly correspond to the managed signature.
         /// </summary>
-        public static unsafe uint MultiplyNoFlags(uint left, uint right, uint* low) => MultiplyNoFlags(left, right, low);
+        public static unsafe uint MultiplyNoFlags(uint left, uint right, uint* low)
+        {
+            var result = MultiplyNoFlags2(left, right); *low = result.Lower; return result.Upper;
+        }
+
+        /// <summary>
+        /// unsigned int _mulx_u32 (unsigned int a, unsigned int b, unsigned int* hi)
+        ///   MULX r32a, r32b, reg/m32
+        /// The above native signature does not directly correspond to the managed signature.
+        /// </summary>
+        internal static unsafe (uint Lower, uint Upper) MultiplyNoFlags2(uint left, uint right) => MultiplyNoFlags2(left, right);
```

Comment:
I know this is an existing pattern for intrinsic, but would be less confusing if it is something like:
`=> throw new NotImplementedException("Intrinsic must be implemented by the runtime.");`
instead of recursive call.

Reply:
I note that the recursiveness is an implementation detail that allows (easy) detection of when the intrinsic must be expanded:

https://github.com/dotnet/runtime/blob/1d352fc8164a43e787cb17782ae26d17e5e28b13/src/coreclr/jit/importer.cpp#L3740-L3741

Reply:
Ah, I see. Thanks for the link. IMO, it would have made the managed APIs more readable if the detection was based off of some descriptive construct.

Reply:
This implementation is actually there to support calls using reflection. It's mostly a theoretical concern, but precludes a need for special handling of reflection for HW intrinsics.

Reply:
However, for internal methods throwing might be a valid option, not sure.